addons/calendar/static/src/core/web/store_service_patch.js

```diff
@@ -0,0 +1,26 @@
+/* @odoo-module */
+
+import { Store } from "@mail/core/common/store_service";
+
+import { deserializeDateTime, formatDateTime } from "@web/core/l10n/dates";
+import { localization } from "@web/core/l10n/localization";
+import { patch } from "@web/core/utils/patch";
+
+/** @type {import("models").Store} */
+const StorePatch = {
+    async onUpdateActivityGroups() {
+        for (const group of Object.values(this.activityGroups)) {
+            if (group.type === "meeting") {
+                for (const meeting of group.meetings) {
+                    if (meeting.start) {
+                        const date = deserializeDateTime(meeting.start);
+                        meeting.formattedStart = formatDateTime(date, {
+                            format: localization.timeFormat,
+                        });
+                    }
+                }
+            }
+        }
+    },
+};
```

Comment:
Could be worth adding super call in case we add some code in original and/or there's another patch

Reply:
Done